src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp

```diff
@@ -180,6 +180,7 @@ real64 HydrofractureSolver::fullyCoupledSolverStep( real64 const & time_n,
     int locallyFractured = 0;
     int globallyFractured = 0;
 
+    // recompute the sparsity pattern at every time step
```

Comment:
we could change the value of that flag each time that there is a topology change. I can go in and implement it  in this PR if you like.

Reply:
Yes, that would be great, feel free to push your modifications to the PR.

Reply:
I thought about this (a long time ago) and came to the conclusion that, given the "decentralized" nature of GEOSX (different components driven by `EventManager` can interact with the mesh and modify it independently at arbitrary times), a more robust system for detecting mesh modifications should be based on some kind of timestamps (whether it be measured in event manager cycles or something else). I.e. any component, such as `SurfaceGenerator`, that modifies the mesh, must also save the timestamp of last modification in the mesh data structure, and every solver must check that timestamp against its own saved timestamp of when the linear system was last rebuilt (and do that for all mesh targets it operates on). Perhaps this warrants an infrastructure discussion?

Reply:
Following up on this morning discussion, I started an implementation of the time stamp (mesh modification time stamp and system setup time stamp). I am not too sure about what to do for the multi-`MeshLevel` case, but the PR works and is ready to be looked at again.